File: uw-count-front-end/src/app/models/bill.ts
import { BillTransaction } from "./bill-transaction";

export interface Bill {
  id: number|null;
  supplierId: number;
  invoiceNo: string;
  date: string;
  dueDate: string;
  subTotal: number;
  discount: number;
  payableBal: number;
  tax: number;
  billTransactions: BillTransaction[];
}

File: uw-count-back-end/src/main/java/org/uwcount/dto/Bill.java
package org.uwcount.dto;

import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Bill {
    private Integer id;
    private Integer supplierId;
    private String invoiceNo;
    private LocalDate date;
    private LocalDate dueDate;
    private Double subTotal;
    private Double discount;
    private Double tax;
    private Double payableBal;
    private List<BillTransaction> billTransactions;
}
